std/assembly/string.ts

```diff
@@ -413,6 +413,52 @@ export class String {
     return out;
   }
 
+  padStart(targetLength: i32, padString: String = changetype<String>(" ")): String {
+    assert(this !== null);
+    var length = this.length;
+    var padLen = padString.length;
+    if (targetLength < length || !padLen) return this;
+    var len = targetLength - length;
+    if (len > padLen) padString = padString.repeat(len / padLen + 1);
+    var out = allocate(targetLength);
+    memory.copy(
+      changetype<usize>(out) + HEADER_SIZE,
+      changetype<usize>(padString) + HEADER_SIZE,
+      <usize>len << 1
+    );
+    if (length) {
+      memory.copy(
+        changetype<usize>(out) + HEADER_SIZE + (<usize>len << 1),
+        changetype<usize>(this) + HEADER_SIZE,
+        <usize>targetLength << 1
+      );
+    }
+    return out;
+  }
+
+  padEnd(targetLength: i32, padString: String = changetype<String>(" ")): String {
+    assert(this !== null);
+    var length = this.length;
+    var padLen = padString.length;
+    if (targetLength < length || !padLen) return this;
+    var len = targetLength - length;
+    if (len > padLen) padString = padString.repeat(len / padLen + 1);
```

Comment:
if `len = 4, padLen = 2`, this would repeat the string `4 / 2 + 1 = 3` times (I'd expect 2 times) - is that correct?

Reply:
Yes. I can improve this for case, when reminder is zero after division